plugin/trino-snowflake/src/main/java/io/trino/plugin/snowflake/SnowflakeClient.java

```diff
@@ -0,0 +1,651 @@
+            Instant timeI = Instant.ofEpochSecond(epoSeconds, adjNano);
+            statement.setString(index, SNOWFLAKE_DATETIME_FORMATTER.format(ZonedDateTime.ofInstant(timeI, zone)));
+        });
+    }
+
+    private static LongWriteFunction snowFlakeDateTimeWriter()
+    {
+        return (statement, index, encodedTimeWithZone) -> {
+            Instant time = Instant.ofEpochMilli(DateTimeEncoding.unpackMillisUtc(encodedTimeWithZone));
+            ZoneId zone = ZoneId.of(DateTimeEncoding.unpackZoneKey(encodedTimeWithZone).getId());
+            statement.setString(index, SNOWFLAKE_DATETIME_FORMATTER.format(time.atZone(zone)));
+        };
+    }
+
+    private static WriteMapping snowFlakeDecimalWriter(Type type)
+    {
+        DecimalType decimalType = (DecimalType) type;
+        String dataType = String.format("decimal(%s, %s)", new Object[] {
+                Integer.valueOf(decimalType.getPrecision()), Integer.valueOf(decimalType.getScale())
+        });
+
+        if (decimalType.isShort()) {
+            return WriteMapping.longMapping(dataType, StandardColumnMappings.shortDecimalWriteFunction(decimalType));
+        }
+        return WriteMapping.objectMapping(dataType, StandardColumnMappings.longDecimalWriteFunction(decimalType));
+    }
+
+    private static LongWriteFunction snowFlakeDateWriter()
+    {
+        return (statement, index, day) -> statement.setString(index, SNOWFLAKE_DATE_FORMATTER.format(LocalDate.ofEpochDay(day)));
+    }
+
+    private static WriteMapping snowFlakeCharWriter(Type type)
+    {
+        CharType charType = (CharType) type;
+        return WriteMapping.sliceMapping("char(" + charType.getLength() + ")",
+                charWriteFunction(charType));
+    }
+
+    private static WriteMapping snowFlakeVarCharWriter(Type type)
+    {
+        String dataType;
+        VarcharType varcharType = (VarcharType) type;
+
+        if (varcharType.isUnbounded()) {
+            dataType = "varchar";
+        }
+        else {
+            dataType = "varchar(" + varcharType.getBoundedLength() + ")";
+        }
+        return WriteMapping.sliceMapping(dataType, StandardColumnMappings.varcharWriteFunction());
+    }
+
+    private static SliceWriteFunction charWriteFunction(CharType charType)
+    {
+        return (statement, index, value) -> statement.setString(index, Chars.padSpaces(value, charType).toStringUtf8());
+    }
+
+    private static WriteMapping snowFlakeTimestampWriter(Type type)
+    {
+        TimestampType timestampType = (TimestampType) type;
+        checkArgument((timestampType.getPrecision() <= SNOWFLAKE_MAX_SUPPORTED_TIMESTAMP_PRECISION),
+                "The max timestamp precision in Snowflake is " + SNOWFLAKE_MAX_SUPPORTED_TIMESTAMP_PRECISION);
+
+        if (timestampType.isShort()) {
+            return WriteMapping.longMapping(
+                    String.format("timestamp_ntz(%d)", new Object[] {Integer.valueOf(timestampType.getPrecision()) }),
```

Comment:
new Object[] { ... } is unnecessary. String.format() is a varargs method.

This should suffice:

```java
String.format("timestamp_ntz(%d)", timestampType.getPrecision())
```